OpenRA.Mods.Common/AI/States/AirStates.cs

```diff
@@ -31,17 +31,19 @@ protected static int CountAntiAirUnits(IEnumerable<Actor> units)
 			var missileUnitsCount = 0;
 			foreach (var unit in units)
 			{
-				if (unit != null && unit.Info.HasTraitInfo<AttackBaseInfo>() && !unit.Info.HasTraitInfo<AircraftInfo>()
-					&& !unit.IsDisabled())
+				if (unit == null || unit.Info.HasTraitInfo<AircraftInfo>())
+					continue;
+
+				var arms = unit.TraitsImplementing<AttackBase>()
+					.Where(ab => !ab.IsTraitPaused && !ab.IsTraitDisabled)
+					.SelectMany(ab => ab.Armaments);
+
+				foreach (var a in arms)
```

Comment:
AttackBase does not necessarily control all the armaments on an actor.  Selecting all armaments ignoring AttackBase would be wrong.

Reply:
Meaning we can have armaments that are not used by any attacking trait?

Reply:
Re: performance/foreach loops: In general I agree that the AI code is probably one of the lower-hanging fruits in terms of performance-wins, but since this PR is primarily about removing IDisable, I think that should be left for later (or a separate PR, in any case).